plugin/src/components/Settings/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect } from 'react'
+import { BsChevronDown } from 'react-icons/bs'
+import { useAtom, useAtomValue } from 'jotai'
+import useAsyncFn from '@/hooks/useAsyncFn'
+import { remixClientAtom } from '@/stores/remixClient'
+import { apiUrl } from '@/utils/network'
+import { solidityVersionAtom, versionsAtom } from '@/atoms'
+import * as Dropdown from '@/ui_components/Dropdown'
+import useTimeoutFn from '@/hooks/useTimeoutFn'
+import './settings.css'
+
+const DEFAULT_DELAY = 5_000
+
+export const Settings = () => {
+  const remixClient = useAtomValue(remixClientAtom)
+  const [solidityVersion, setSolidityVersion] = useAtom(solidityVersionAtom)
+  const [versions, setVersions] = useAtom(versionsAtom)
+
+  const [, refetchVersions] = useAsyncFn(async () => {
```

Comment:
with this approach, we are loading versions only when rendering "Settings" tab, so each time we need to go here and wait until plugin fetch it, but it should done as soon as plugin loads

Reply:
Ah good catch!!. Thank you. I moved it to CompilerVersion component so it loads on initial render.